GeometricSeqPhenotype.java

```diff
@@ -0,0 +1,316 @@
+     * @effects Constructs a new GeometricSeqPhenotype that has a starting sequence
+     * and GeometricPhenotype with the data content of the given parameters.
+     */
+    public GeometricSeqPhenotype(double tA, double tB, char[] startingSequence) {
+        this();
+        this.traitA = tA;
+        this.traitB = tB;
+        this.nucleotideSequence = startingSequence;
+        checkRep();
+    }
+
+    /**
+     * Constructor that creates a new GeometricSeqPhenotype.
+     *
+     * @param tA the x-coordinate of the new GeometricSeqPhenotype.
+     * @param tB the y-coordinate of the new GeometricSeqPhenotype.
+     * @param startingSequence the starting nucleotide sequence of the GeometricSeqPhenotype to be constructed.
+     * @param e the number of epitope mutations this GeometricSeqPhenotype is from startingSequence
+     * @param nE the number of non-epitope mutations from startingSequence is from startingSequence
+     * @requires nucleotideSequence != null && nucleotideSequence.length() > 0 && nucleotideSequence.length() % 3 == 0
+     * @effects Constructs a new GeometricSeqPhenotype that has a starting sequence
+     * and GeometricPhenotype with the data content of the given parameters.
+     */
+    public GeometricSeqPhenotype(double tA, double tB, char[] startingSequence, int e, int nE) {
+        this.traitA = tA;
+        this.traitB = tB;
+        this.nucleotideSequence = startingSequence;
+        this.eMutation = e;
+        this.neMutation = nE;
+        checkRep();
+    }
+
+    /**
+     * Return x-coordinate of where this GeometricSeqPhenotype is in Euclidean space
+     *
+     * @return position of this GeometricSeqPhenotype along the x-axis
+    */
+    public double getTraitA() {
+        return traitA;
+    }
+
+    /**
+     * Return y-coordinate of where this GeometricSeqPhenotype is in Euclidean space
+     *
+     * @return position of this GeometricSeqPhenotype along the y-axis
+     */
+    public double getTraitB() {
+        return traitB;
+    }
+
+    /**
+     * Returns the nucleotide sequence of this GeometricSeqPhenotype.
+     * Valid example outputs include "ACG" and "ACGTGTACGTGT"
+     *
+     * @return the nucleotide sequence of the GeometricSeqPhenotype represented by this.
+     */
+    public String getSequence() {
+        return String.valueOf(this.nucleotideSequence);
+    }
+
+    /**
+     * Return the (Euclidean) distance between this GeometricSeqPhenotype and p in Euclidean space
+     *
+     * @param p  GeometricSeqPhenotype to calculate the distance between
+     * @return distance between this GeometricSeqPhenotype and p
+     */
+    public double distance(Phenotype p) {
+        GeometricSeqPhenotype p2d = (GeometricSeqPhenotype) p;
+        double distA = (getTraitA() - p2d.getTraitA());
+        double distB = (getTraitB() - p2d.getTraitB());
+        double dist = (distA * distA) + (distB * distB);
+        dist = Math.sqrt(dist);
+        checkRep();
```

Comment:
Sanity checks are good, but could we move it a little farther out in the code, when the condition could be violated?